modules/@apostrophecms/attachment/index.js

```diff
@@ -294,7 +296,10 @@ module.exports = {
           }
         }
         info.used = true;
-        await self.db.updateOne({ _id: info._id }, info);
+
+        await self.db.updateOne({ _id: info._id }, {
```

Comment:
Oops, my bad, I forgot that `replaceOne` is a better replacement for updateOne here. It does *not* require `$set`, which is good because `$set` would not remove properties that should no longer exist.